analyzers/src/SonarAnalyzer.CSharp/Rules/PrivateFieldUsedAsLocalVariable.cs

```diff
@@ -184,12 +184,12 @@ bool IsPrecededWithWrite(SyntaxNode readStatementOrArrowExpression)
                     }
 
                     bool IsOverwritingValue(StatementSyntax statement) =>
-                        writeStatements.Contains(statement) &&
-                        !readStatements.Contains(statement);
+                        writeStatements.Contains(statement)
```

Comment:
Function could be made static

Reply:
These two methods use fields from the class, and I would not make them static and forward the fields as parameters. What was your idea, how to transform them to static methods?

Reply:
Oh, my bad.